### PR TITLE
chore: add TSDocs to the API Key Module

### PR DESCRIPTION
**Pull request by shahednasser**
Add TSDocs to the API Key Module's resources

### CODE REVIEW

File: packages/types/src/api-key/common/api-key.ts [hunk 1]
```diff
@@ -1,23 +1,90 @@
...
+   */
   created_by: string
+
+  /**
+   * The date the API key was created.
+   */
   created_at: Date
+
+  /**
+   * Who revoked the API key.
```
Comment (sradevski) [inline packages/types/src/api-key/common/api-key.ts]: NIT: I think it's nice to specify what exactly this points to (eg. "the ID of the user that revoked the API key")

File: packages/types/src/api-key/mutations/api-key.ts [hunk 2]
```diff
@@ -1,24 +1,74 @@
...
   id?: string
+
+  /**
+   * The title of the API key.
+   */
   title?: string
+
+  /**
+   * The type of the API key. It's only
+   * usable and required when creating an API key.
```
Comment (sradevski) [inline packages/types/src/api-key/mutations/api-key.ts]: Can we rephrase this to `Required only when creating an API key`? I think the usable part is implied

**Review by sradevski — APPROVED** [review]

File: packages/types/src/api-key/common/api-key.ts (hunk 1 above)
Comment (shahednasser) [reply packages/types/src/api-key/common/api-key.ts]: I was thinking of that too, but also felt like it's somewhat limiting. i.e. if someone is using the module on their own, they can use something other than user ID such as email. However, it makes sense to add an example so I'll do that 👍🏻
